### DISCUSSION TITLE
Exclude packages

### DISCUSSION THREAD

**Discussion by RafalFilipek** [commonalityco/commonality#328] [Ideas]
Hi,

I'm currently integrating commonality into our project, and I've noticed in a few places that it would be useful to have the option to exclude packages for checks. The current configuration only allows for specifying "positive" checks. This would probably require some extension.

In general, this is for cases where 99% of packages should work the same way, but a few should not be checked.

I think it would be useful to have a way to exclude packages from checks so that we can avoid having to create separate configurations for different cases, or adding tags to almost all packages.

Maybe we can update the `checks` option to accept something like this:

```
{
  'include': ['foo', 'bar'] // or '*'
  'exclude': ['biz']
  'checks': ['a', 'b', 'c']
}
```

Thanks,

Rafał

**Comment by alec-chernicki** [comment]
@RafalFilipek sorry for the delay in gettig back to you! this is really interesting, right now the only way to achieve this is to add tags to the packages you want to target. this choice was made to incentivizes you to start tagging packages because tags will also form the basis of more future functionality. 

this is really interesting and I definitely want to learn more so we can modify the behavior of selectors if needed. could you provide some use cases for applying checks to 99% of packages in your project?

**Comment by RafalFilipek** [comment]
Hey,

For example:

- I have few packages without code. With assets / configs etc. I would rather not use `has-jest-config`, `has-ts-config` etc. rules there.
- I have some deprecated / lecagy packages. In that case, `legacy`, `no-check` tag could be useful.

**Reply by alec-chernicki** [reply]
That makes sense! Would adding negation functionality (`!`) to [selectors](https://docs.commonality.co/selectors) help with this?

I.e. if you want to select everything *except* config packages. You'd tag those packages with something like `config` then in your project configuration file you'd do something like:

.commonality/config.json
```json
{
  "checks": {
    "!config": ["my-check"]
  }
}
```

**Reply by mrmckeb** [reply]
I think there are still specific cases where you want to apply a check to everything except one specific package.

For example, I may want to check that every package has a command to check types, except for my shared TypeScript config package - as that's just a JSON file.
